src/base/language_support.c

```diff
@@ -3789,7 +3789,7 @@ lang_strmatch_utf8_uca_w_level (const COLL_DATA * coll_data, const int level, bo
 	  do
 	    {
 	      w2 = GET_UCA_WEIGHT (level, ce_index2, uca_w_l13_2, uca_w_l4_2);
-	      if (w2 != 0)
+	      if (w2 != 0 || !ignore_trailing_space)
```

Comment:
why `!ignore_trailing_space` is added? `w2 != 0` is enough to check for non-zero weight and return it strings are not equal.

Reply:
although the weight is zero, check it is ignore_trailing_space is false because trailing space (zero weight) should not be ignored.

Reply:
There may exist a case that `!ignore_trailing_space` is true and w2 is 0 but str2 is not space.

Reply:
because the string1's length is already less than string2's, so the comparing is done as string1 < string2.

Reply:
That means there are remaining chars in string2, this routine consumes trailing zero-weight values.
"space" is zero-weight. but I'm not sure that all zero-weight values are from "space".

Reply:
Once the trailing character exists, whatever it is, the comparing is done although the weight is zero but not SPACE.
The trailing space's weight zero and the null character is zero weight but the string length is more matter than its weight.